### PR TITLE
RESTEASY-2161 remove @NoJackson annotation

### PR DESCRIPTION
**Pull request by liweinan**
- remove @NoJackson annotation (because jettison is removed)
- clean Jettison related tests
- update documents

### CODE REVIEW

File: docbook/reference/en/en-US/modules/Jaxb.xml [hunk 1]
```diff
@@ -2,7 +2,10 @@
    <title>JAXB providers</title>
 
    <para>
-
+      NOTE: This section contains some outdated information about the deprecated Jettison provider, and the relative information will be
+      replaced with the usage of currently supported Jackson2 provider.
+   </para>
+   <para>
```
Comment (ronsigal) [inline docbook/reference/en/en-US/modules/Jaxb.xml]: Hi @liweinan,

Now that you have removed references to jettison, this section DOESN'T have outdated information any more. I would just say that the jettison provider is no longer supported.

File: docbook/reference/en/en-US/modules/Jaxb.xml [hunk 2]
```diff
@@ -471,7 +462,6 @@ public void newBook(@BadgerFish Book book) {...}
 
       </para>
       <para>
-
          The default Jettison Mapped Convention would return JSON that looked like this:
```
Comment (ronsigal) [inline docbook/reference/en/en-US/modules/Jaxb.xml]: This should go, right?

Comment (ronsigal) [reply docbook/reference/en/en-US/modules/Jaxb.xml]: See
```
<programlisting>
java.lang.IllegalStateException: Invalid JSON namespace: http://jboss.org/books
at org.codehaus.jettison.mapped.MappedNamespaceConvention.getJSONNamespace(MappedNamespaceConvention.java:151)
at org.codehaus.jettison.mapped.MappedNamespaceConvention.createKey(MappedNamespaceConvention.java:158)
at org.codehaus.jettison.mapped.MappedXMLStreamWriter.writeStartElement(MappedXMLStreamWriter.java:241)
</programlisting>
```

**Comment by ronsigal** [comment]
Hey @liweinan,

Good job. I just have a couple of comments.

-Ron

**Comment by liweinan** [comment]
@ronsigal Thanks for reviewing Ron! I'll remove the outdated parts of doc after coming back from holiday :D

**Comment by ronsigal** [comment]
Thanks, @liweinan. Have a good holiday.

File: docbook/reference/en/en-US/modules/Jaxb.xml (hunk 1 above)
Comment (liweinan) [reply docbook/reference/en/en-US/modules/Jaxb.xml]: let me remove this note. And the deprecation of the jettison provider I guess should be announced in release note. In addition it's written in our blog post: https://developer.jboss.org/en/resteasy/blog/2019/02/12/jettison-has-been-removed-on-master-branch

File: docbook/reference/en/en-US/modules/Jaxb.xml (hunk 2 above)
Comment (liweinan) [reply docbook/reference/en/en-US/modules/Jaxb.xml]: yes, the @BadgerFish things are part of jettison providers, let me remove them.

**Comment by liweinan** [comment]
@ronsigal This branch needs rebase, and I have created a new PR for it(in case I mass things up): https://github.com/resteasy/Resteasy/pull/1878

**Pull request closed without merging**